src/insexpand.c

```diff
@@ -553,20 +553,22 @@ ins_compl_infercase_gettext(
 	return IObuff;
 
     p = str;
-    for (i = 0; i < char_len; ++i)
-	if (has_mbyte)
+    if (has_mbyte)
+    {
+	for (i = 0; i < char_len; ++i)
 	    wca[i] = mb_ptr2char_adv(&p);
-	else
+    }
+    else
+    {
+	for (i = 0; i < char_len; ++i)
 	    wca[i] = *(p++);
+    }
```

Comment:
that doesn't look like a clear improvement.

Reply:
has_mbyte only needs to be checked once. Similar code can be found in find_word_end. Consider the usage scenario of has_mbyte. Maybe a macro can be used to wrap this condition?

Reply:
True, but  I found readability more important than improving the efficiency of having to check this variable only once.